idact/detail/allocate_resources_app/main.py

```diff
@@ -0,0 +1,144 @@
+from idact.detail.config.client.client_cluster_config import ClusterConfigImpl
+from idact.detail.helper.ensure_stdin_has_fileno import \
+    ensure_stdin_has_fileno
+from idact.detail.jupyter_app.app_allocation_parameters import \
+    AppAllocationParameters
+from idact.detail.jupyter_app.format_allocation_parameters import \
+    format_allocation_parameters
+from idact.detail.jupyter_app.native_args_conversion import \
+    convert_native_args_from_command_line_to_dict
+from idact.detail.jupyter_app.override_parameters_if_possible import \
+    override_parameters_if_possible
+from idact.detail.log.get_logger import get_logger
+
+SNIPPET_SEPARATOR_LENGTH = 10
+
+
+@click.command()
+@click.argument('cluster_name',
+                type=str)
+@click.option('--environment', '-e',
+              default=None,
+              type=str,
+              help="Environment path. Default: ~/.idact.conf"
+                   " or the value of IDACT_CONFIG_PATH.")
+@click.option('--save-defaults',
+              is_flag=True,
+              help="Save allocation parameters as defaults for next time.")
+@click.option('--reset-defaults',
+              is_flag=True,
+              help="Reset unspecified allocation parameters to defaults.")
+@click.option('--nodes',
+              default=None,
+              type=int,
+              help="Cluster node count. [Allocation parameter]."
+                   " Jupyter notebook will be deployed on the first node."
+                   " Default: 1.")
+@click.option('--cores',
+              default=None,
+              type=int,
+              help="CPU core count per node. [Allocation parameter]."
+                   " Default: 1")
+@click.option('--memory-per-node',
+              default=None,
+              type=str,
+              help="Memory per node. [Allocation parameter]."
+                   " Default: 1GiB")
+@click.option('--walltime',
+              default=None,
+              type=str,
+              help="Maximum time to allocate"
+                   " the resources for. Format: [days-]hours:minutes:seconds."
+                   " [Allocation parameter]."
+                   " Default: 0:10:00")
+@click.option('--native-arg',
+              nargs=2,
+              multiple=True,
+              type=str,
+              metavar='ARG VALUE',
+              default=[],
+              help="Native arguments for the workload manager."
+                   " Flags have value=None, e.g. --native-arg --flag None."
+                   " Can be repeated for multiple native args:"
+                   " --native-arg -arg1 v1 --native-arg -arg2 v2 (...)."
+                   " Values are not validated."
+                   " Supported arguments take precedence over native"
+                   " arguments."
+                   " Arguments specified later override earlier arguments."
+                   " [Allocation parameter]."
+                   " Default: No native arguments.")
+def main(cluster_name: str,
```

Comment:
If this should provide allocation for the future, shouldn't be also --begin argument?

Reply:
It should be passed as a native argument

Reply:
Maybe there should be something in help? Like `Allocating for the future could be done by passing the special argument to the native args f.eg. '--begin'`